### ISSUE TITLE
PDF Viewer opens but no content displayed

### ISSUE THREAD

**Issue by CSchaefer70** [nextcloud/files_pdfviewer#281]
NextCloud version 20.0.4 (but confirmed issue on 20.0.2 and 20.0.3).  Version 17 works fine.  Fresh install from https://nextcloud.com/install/#instructions-server

On shared host server BlueHost.  PHP 7.3

To replicate simply click on any PDF to view.  The Viewer opens but there is no content displayed.  Confirmed on Edge, Firefox and Chrome.  The PDF file may be downloaded and will open using a local app like Acrobat.

**Comment by skjnldsv** [comment]
Hello!
I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by CSchaefer70** [comment]
So it looks like a problem with core build?  I did try a fresh install and the problem is not present before v20.

Here is the browser console log and a screen video:

08:40:45.708 viewer.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
08:40:51.850 /nc21/index.php/core/preview?fileId=5737&x=32&y=32:1 Failed to load resource: the server responded with a status of 404 ()
08:40:53.316 /nc21/index.php/core/preview?fileId=5765&x=32&y=32:1 Failed to load resource: the server responded with a status of 404 ()
08:40:53.935 /nc21/index.php/core/preview?fileId=5765&c=1f6e5e11d6c205b03a3def624b3f7d88&x=375&y=375&forceIcon=0:1 Failed to load resource: the server responded with a status of 404 ()
08:40:54.952 /nc21/index.php/core/preview?fileId=5736&x=32&y=32:1 Failed to load resource: the server responded with a status of 404 ()
08:41:03.854 /nc21/index.php/apps/files_pdfviewer/?file=https%3A%2F%2Fndl360life.com%2Fnc21%2Fremote.php%2Fdav%2Ffiles%2Fadmin%2FReasons%20to%20use%20Nextcloud.pdf:1 Failed to load resource: the server responded with a status of 406 ()
08:41:16.909 globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
ie @ globals.js:60
08:41:16.909 globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
ie @ globals.js:60
08:42:27.751 ​ jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
ie @ ​
get @ ​
updateTopMenu @ ​

https://user-images.githubusercontent.com/26978180/104022512-48137880-51b8-11eb-9eb5-a77a4ef58e38.mp4

**Comment by skjnldsv** [comment]
> 08:41:03.854 /nc21/index.php/apps/files_pdfviewer/?file=https%3A%2F%2Fndl360life.com%2Fnc21%2Fremote.php%2Fdav%2Ffiles%2Fadmin%2FReasons%20to%20use%20Nextcloud.pdf:1 Failed to load resource: the server responded with a status of 406 ()

This is your issue :+1: 
Can you open the url directly and check the response content of this request?

**Comment by CSchaefer70** [comment]
Well this is a change... I attempted to open a PDF today and the preview GUI is quite different... please view the screen video.  Did my shared server have some cache stuck or corrupt?  I don't follow what happened!  It's opening now yet I did nothing?

https://user-images.githubusercontent.com/26978180/104177204-65be2900-53d6-11eb-8aca-15754c5e3025.mp4

**Comment by szaimen** [comment]
@CSchaefer70 did you just install the ebook reader app?
https://apps.nextcloud.com/apps/epubreader

**Comment by CSchaefer70** [comment]
Yikes... u r correct! I forgot!  I have been trying so many things.  So yes the PDF is now running thru the EPub Reader.

So when I disable the preview goes back to the original issue of not opening content. (406 error).  But isn't that a browser issue not accepting specific characters or header? I tried on several different OS (Win8, Win10, Linux) with the same result.  So it must be something with my server??  I'm not sure where to look as it is a "Shared Host".

![2021-01-11_6-38-26](https://user-images.githubusercontent.com/26978180/104178484-68218280-53d8-11eb-9a4d-9c3199895875.jpg)

**Comment by szaimen** [comment]
@CSchaefer70 I guess the next step is to do what @skjnldsv has suggested:
https://github.com/nextcloud/files_pdfviewer/issues/281#issuecomment-757732623

**Comment by CSchaefer70** [comment]
Hmmm... like via webDav client?  Otherwise, I think I would need to put in an exception on the .htaccess file??  The file is valid and it does open with the epub app.  Is it possible to use the "occ" command on a shared server? I do have some error warnings which I have not been able to clean-up (see attached).   Some warnings, like "op cache" I cannot fix because the installation is on a shared host.  It's never bothered me in the past so I never resolved.   I guess this may be beyond my knowledge.  I do have access to the DB via phpMyAdmin but have not been success making changes (due to my limit use of the tool).

![2021-01-11_7-05-54](https://user-images.githubusercontent.com/26978180/104180954-fb0fec00-53db-11eb-80d3-03a73482f510.jpg)

**Comment by szaimen** [comment]
@skjnldsv ⤴ 🤔

**Comment by CSchaefer70** [comment]
Yeah, I know.  Just doing the best I can.  For now I'll keep using the epub app unless anyone can educate me with a big stick!

**Comment by szaimen** [comment]
@CSchaefer70 I guess what @skjnldsv suggested was that you open https%3A%2F%2Fndl360life.com%2Fnc21%2Fremote.php%2Fdav%2Ffiles%2Fadmin%2FReasons%20to%20use%20Nextcloud.pdf and check the response content of this request.

**Comment by CSchaefer70** [comment]
So I guess you're asking to use "https://ndl360life.com/nc21/remote.php/dav/files/admin/Reasons/to/use/Nextcloud.pdf" as a URL?  

This responds the same... "406"

I tried a simple filename without "/"... just renamed "Reason to use Nextcloud.pdf" to "Nextcloud.pdf".  No different; pdf viewer opens but no content.

If I try to access the webDAV "admin" location [_**https://ndl360life.com/nc21/remote.php/dav/files/admin/**_] I receive notice that I can ONLY use a client.

"**This is the WebDAV interface. It can only be accessed by WebDAV clients such as the Nextcloud desktop sync client**."

![2021-01-13_16-19-56](https://user-images.githubusercontent.com/26978180/104512174-cd819900-55bb-11eb-94dd-a9cb9c6a37f3.jpg)

**Comment by skjnldsv** [comment]
Sorry, was on vacations!
@CSchaefer70 please fix your setup issues (screenshot).


> This responds the same... "406"

Can you share your nextcloud log? See if there are any errors in it? Because I have never seen a "HTTP 406 Not Acceptable" here. :thinking:

**Comment by CSchaefer70** [comment]
Hey thx for your interest.  I gave up and use the "EPUB/CBZ/PDF ebook Reader".  However, because u asked I disabled the ebook reader and tried again to open a PDF with the same result... the PDF attempts to open but no display... dark screen.  I looked at the logging and see nothing associated.  I just installed the OCC Web app and cleaned up some of my table error warnings which show up in Admin "Overview"!!

Anyway, if you want to poke around I don't have anything private on this site so I made u and Admin account: 

URL = https://ndl360life.com/nc21/index.php/login
u = JohnM
p = JohnM2021!

Thanks again for your interest... I'm finding NextCLoud more and more useful, but, recently the upgrades are driving me nuts.  MySQL is version 5.6 and I just read I need version 8+ to upgrade to NCHub version 21.
